truss-chains/truss_chains/framework.py

```diff
@@ -241,22 +370,32 @@ def _validate_and_describe_endpoint(
     * Generators are allowed, too (but not yet supported).
     """
     if not hasattr(cls, definitions.ENDPOINT_METHOD_NAME):
-        raise definitions.ChainsUsageError(
-            f"`{cls.name}` must have a {definitions.ENDPOINT_METHOD_NAME}` method."
+        _collect_error(
+            f"Chainlets must have a `{definitions.ENDPOINT_METHOD_NAME}` method.",
+            _ErrorKind.MISSING_API_ERROR,
+            location,
+        )
+        return definitions.EndpointAPIDescriptor(
+            input_args=[], output_types=[], is_async=False, is_generator=False
         )
+
     # This is the unbound method.
     endpoint_method = getattr(cls, definitions.ENDPOINT_METHOD_NAME)
+
+    line = inspect.getsourcelines(endpoint_method)[1]
+    location = location.model_copy(
+        update={"line": line, "method_name": definitions.ENDPOINT_METHOD_NAME}
+    )
+
     if not inspect.isfunction(endpoint_method):
-        raise definitions.ChainsUsageError(
-            f"`{cls.name}.{definitions.ENDPOINT_METHOD_NAME}` must be a method."
-        )
+        _collect_error("`Endpoints must be a method.", _ErrorKind.TYPE_ERROR, location)
```

Comment:
Should we return here? I guess I'm generally wondering if it's safe to collect error without early return/break since removing `raise` changes the control flow.

Reply:
Good point. Thanks for being thorough!